packages/contracts-bedrock/src/L1/OptimismPortal.sol

```diff
@@ -360,6 +472,38 @@ contract OptimismPortal is Initializable, ResourceMetering, ISemver {
         public
         payable
         metered(_gasLimit)
+    {
+        (address token,) = gasPayingToken();
+        if (token != Constants.ETHER && msg.value != 0) {
+            revert NoValue();
+        }
```

Comment:
I wonder if there's benefit in a future where we stop supporting native ETH, and take a uniswap-like approach of wrapping to WETH before deposit and unwrapping on L2, with the idea that wrapping happens in a peripheral contract to keep the core code slim